pkg/provider/provider.go

```diff
@@ -14,11 +14,22 @@
 
 package provider
 
-import "context"
+import (
+	"context"
+
+	"github.com/bank-vaults/secret-init/pkg/common"
+)
 
 // Provider is an interface for securely loading secrets based on environment variables.
 type Provider interface {
+	// GetName returns the name of the provider
+	GetName() string
+	// NewProvider creates a new provider
+	NewProvider(ctx context.Context, appConfig *common.Config) (Provider, error)
+	// LoadSecrets loads secrets from the provider based on the given paths
 	LoadSecrets(ctx context.Context, paths []string) ([]Secret, error)
+	// Valid returns whether a env value can be handled by the provider
+	Valid(envValue string) bool
 }
```

Comment:
we should remove the factory from the interface itself as it can introduce a lot of weird behaviour

```suggestion
type ProvideType string

type ProviderFactory func(ctx context.Context, cfg *common.Config) (Provider, error)

// Provider is an interface for securely loading secrets based on environment variables.
type Provider interface {
	// Type returns the type of the provider (e.g. vault, aws)
	Type() ProvideType
	// LoadSecrets loads secrets from the provider based on the given paths
	LoadSecrets(ctx context.Context, paths []string) ([]Secret, error)
	// Valid returns whether a env value can be handled by the provider
	Valid(envValue string) bool
}
```